lib/kernel/src/erl_epmd.erl

```diff
@@ -291,26 +286,26 @@ init(_) ->
 
 handle_call({register, Name, PortNo, Family}, _From, State) ->
     case State#state.socket of
-	P when P < 0 ->
-	    case do_register_node(Name, PortNo, Family) of
-		{alive, Socket, Creation} ->
-		    S = State#state{socket = Socket,
-				    port_no = PortNo,
-				    name = Name,
-				    family = Family},
-		    {reply, {ok, Creation}, S};
+        P when P < 0 ->
+            case do_register_node(Name, PortNo, Family) of
+                {alive, Socket, Creation} ->
+                    S = State#state{socket = Socket,
+		                    port_no = PortNo,
+		                    name = Name,
+		                    family = Family},
+                    {reply, {ok, Creation}, S};
                 Error ->
-                    case init:get_argument(erl_epmd_port) of
+                    case erl_epmd_listen_port() of
                         {ok, _} ->
                             {reply, {ok, -1}, State#state{ socket = -1,
                                                            port_no = PortNo,
                                                            name = Name} };
```

Comment:
Making `Port > 0` is an incompatible change. What is the reason for checking that `Port > 0`?

Reply:
Yes, since you are ok with moving it to `maint`, I won't change this in this PR!

---

I can still send a separate PR against `main` to check `Port > 0`, if you agree it makes sense.

Let me rephrase the above reasoning. This conditional says: it doesn't matter if epmd is running as long as `erl_epmd_listen_port` is set, because in that case we know which node port to talk to. However, if the user sets `erl_epmd_listen_port` to `0`, then we actually don't know the node port. Therefore, the proposed change is to only ignore epmd when the port is non-zero.

And to give another reason, the default behaviour of `erl_epmd` is as if `erl_epmd_listen_port = 0`, but in this code path setting `erl_epmd_listen_port = 0` explicitly does make a difference.